tests/test_nominatim_gazetteer.py

```
@@ -0,0 +1,215 @@
+        gz = ng.NominatimGazetteer()
+        with pytest.raises(GazetteerAmbiguousError) as exc_info:
+            gz.lookup("Birmingham", country_hint="US")
+        assert "country_hint='US'" in str(exc_info.value)
+
+    def test_unknown_raises_not_found(self, fake_client):
+        fake_client.geocode.return_value = []
+        gz = ng.NominatimGazetteer()
+        with pytest.raises(GazetteerNotFoundError):
+            gz.lookup("Atlantis")
+
+    def test_empty_name_raises(self, fake_client):
```

Ruff (0.15.12)

[warning] Unused method argument: `fake_client`

(ARG002)